sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/LogDivertAppender.java

```diff
@@ -276,12 +276,19 @@ private static StringLayout initLayout(OperationLog.LoggingLevel loggingMode) {
     return getLayout(isVerbose, layout);
   }
 
-  public LogDivertAppender(OperationManager operationManager,
-    OperationLog.LoggingLevel loggingMode) {
+  public static LogDivertAppender create(OperationManager operationManager,
+                                         OperationLog.LoggingLevel loggingMode) {
+    CharArrayWriter writer = new CharArrayWriter();
+    return new LogDivertAppender(operationManager, loggingMode, writer);
+  }
+
+  private LogDivertAppender(OperationManager operationManager,
```

Comment:
Any reason to change it to private constructor and add `create` for constructing?

Reply:
we need CharArrayWriter  as parameter pass to  LogDivertAppender constructor , then pass to supper constructor.